app/src/main/java/org/simple/clinic/teleconsultlog/prescription/medicines/TeleconsultMedicineItem.kt

```diff
@@ -0,0 +1,77 @@
+package org.simple.clinic.teleconsultlog.prescription.medicines
+
+import androidx.recyclerview.widget.DiffUtil
+import io.reactivex.subjects.Subject
+import kotlinx.android.synthetic.main.list_item_teleconsult_medicine.*
+import org.simple.clinic.R
+import org.simple.clinic.drugs.PrescribedDrug
+import org.simple.clinic.teleconsultlog.medicinefrequency.MedicineFrequency
+import org.simple.clinic.widgets.ItemAdapter
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.recyclerview.ViewHolderX
+import org.simple.clinic.widgets.visibleOrGone
+
+data class TeleconsultMedicineItem(
+    val prescribedDrug: PrescribedDrug,
+    val isLastItem: Boolean
+) : ItemAdapter.Item<UiEvent> {
```

Comment:
Please separate adapter events from the `UiEvent` interface? What we generally do is declare the screen specific events in its own Mobius event type and then have separate events for the adapter items and then map them at the screen level.